flink-table-store-core/src/main/java/org/apache/flink/table/store/file/catalog/FileSystemCatalog.java

```diff
@@ -162,6 +162,11 @@ public void createTable(Identifier identifier, UpdateSchema table, boolean ignor
         uncheck(() -> new SchemaManager(path).commitNewVersion(table));
     }
 
+    @Override
+    public void renameTable(Identifier fromTable, Identifier toTable, boolean ignoreIfNotExists) {
+        throw new UnsupportedOperationException("Cannot rename FileSystem catalog tables");
```

Comment:
Is there any reason to not support renaming here?

Reply:
My initial idea is that some file systems (such as s3) do not support rename operation, so I did not add the file system rename operation.

I see that the latest interface(`org.apache.flink.table.store.fs.FileIO#rename`) adds the rename operation, so I also add the file system catalog rename operation. If some filesystem does not support the rename operation,  rename table will fail.